### DISCUSSION TITLE
ffmpeg error when generating APNG after updating to new Linux Mint version, temporary files not saved

### DISCUSSION THREAD

**Discussion by dpitch40** [ImageMagick/ImageMagick#7052] [Help]
### ImageMagick version

7.1.1-26

### Operating system, version and so on

Linux Mint 21.2, kernel version 5.15.0-91-generic

### Description

I am generating an animated image from a series of still images. (PNGs) The command I am using to generate it looks like this: 

```~/Downloads/AppImages/magick -loop 0 -dispose background -delay 12 Frame1.png -dispose background -delay 8 Frame2.png ...(many more frames)... -crop 320x320+79+114 +repage -resize 160x160 Output.apng```

There are about 40 frames (input images) total. This command works if I output to a .gif file, but is failing if I output to a .apng file. The error message is `ffmpeg: symbol lookup error: /lib/x86_64-linux-gnu/libgobject-2.0.so.0: undefined symbol: g_uri_ref` If I run ImageMagick in verbose mode, I see this command is the one that is failing:

```'ffmpeg' -nostdin -loglevel error -i '/tmp/magick-FwW9sfQeUgItH1xx4lOOHnnamHHv9T1r%d.webp' -plays 0 '/tmp/magick-TLfTmVJ2vuJvJiGMHdyVXIqWtHauRLVm.APNG'```

I further see that there are no longer any magic- files in my /tmp directory, even though the script failed. So I have two questions:

1. Before I ask ffmpeg support about this, I would like to try running the failing command directly to confirm, but I can't because the temporary files it's being run on are being deleted. Why are they being deleted if the script failed? Is there any way to prevent their deletion?
2. Does anyone here have any idea why ffmpeg is failing? It is version 4.4.2-0ubuntu0.22.04.1, installed via a normal `sudo apt-get` command. I only started seeing this after I updated from Linux Mint 19.x to 21.2.

**Comment by urban-warrior** [comment]
Do a search for `undefined symbol: g_uri_ref`.  Fix the issue and your command should execute without complaint.

**Reply by dpitch40** [reply]
The various answers I get from the top 6 results from this search (on DuckDuckGo) are:

- [Set up a Python virtual environment, or install libglib2.0-0](https://stackoverflow.com/questions/65693394/why-am-i-getting-the-importerror-lib-libgobject-2-0-so-0-undefined-symbol-g) (this is not happening within Python code for me, and I already have libglib2.0-0 2.72.4-0ubuntu2.2 installed)
- [Set the appimage to be executable, or make sure the versions of libgobject-2.0.so.0 and libglib-2.0.so.0 being used match](https://github.com/tenacityteam/tenacity-legacy/issues/723) (my AppImage is set to executable; about libglib-2.0.so.0, see the next answer)
- [Unpack the appimage, remove usr/lib/libglib-2.0.so.0 from it, repackage, and run the new appimage](https://github.com/sharkwouter/oceanpop/issues/57) (tried this, did not resolve the issue)
- [Julia-specific discussion that doesn't appear pertinent to my issue](https://discourse.julialang.org/t/glmakie-libgobject-2-0-so-error/97850/11)
- [Someone having a similar issue, but no answers posted](https://www.solveforum.com/forums/threads/solved-opencv-libgobject-2-0-so-0-undefined-symbol-g_uri_ref.1022351/)
- [Get rid of libgio-2.0.so.0](https://github.com/opencv/opencv/issues/20212) (the ImageMagick appimage doesn't contain this library)

Do you have any more specific suggestions on what I should be searching for or trying?
